src/Middleware/RateLimiting/src/RateLimitingMiddleware.cs

```diff
@@ -149,11 +149,19 @@ private LeaseContext CombinedAcquire(HttpContext context)
                 return new LeaseContext() { GlobalRejected = false, Lease = endpointLease };
             }
         }
-        catch (Exception)
+        catch (Exception ex)
         {
             endpointLease?.Dispose();
             globalLease?.Dispose();
-            throw;
+            if (ex is TaskCanceledException)
```

Comment:
Look for `OperationCanceledException` and check if `context.HttpContext.RequestAborted.IsCancellationRequested` is true.

Reply:
As in only log if it's an `OperationCanceledException` _and_ `IsCancellationRequested` is true (otherwise throw)?

Reply:
Right